packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -452,12 +435,57 @@ export async function sourceNodes(
           counter++
         }
 
-        nodeToUpdate.internal.contentDigest = `${initialContentDigest}${CONTENT_DIGEST_COUNTER_SEPARATOR}${counter}`
-        createNode(nodeToUpdate)
+        const newNode = {
+          ...nodeToUpdate,
+          internal: {
+            ...nodeToUpdate.internal,
+            // We need to remove properties from existing fields
+            // that are reserved and managed by Gatsby (`.internal.owner`, `.fields`).
+            // Gatsby automatically will set `.owner` it back
+            owner: undefined,
+            // We add or modify counter postfix to contentDigest
+            // to make sure Gatsby treat this as data update
+            contentDigest: `${initialContentDigest}${CONTENT_DIGEST_COUNTER_SEPARATOR}${counter}`,
+          },
+          // `.fields` need to be created with `createNodeField` action, we can't just re-add them.
+          // Other plugins (or site itself) will have opportunity to re-generate them in `onCreateNode` lifecycle.
+          // Contentful content nodes are not using `createNodeField` so it's safe to delete them.
+          // (Asset nodes DO use `createNodeField` for `localFile` and if we were updating those, then
+          // we would also need to restore that field ourselves after re-creating a node)
+          fields: undefined, // plugin adds node field on asset nodes which don't have reverse links
+        }
+
+        // memory cached nodes are mutated during back reference checks
+        // so we need to carry over the changes to the updated node
+        if (node.__memcache) {
+          for (const key of Object.keys(node)) {
+            if (!key.endsWith(`___NODE`)) {
+              continue
+            }
+
+            newNode[key] = node[key]
+          }
+        }
+
+        createNode(newNode)
+
+        if (existingNodesLoopCount++ % 2000 === 0) {
+          // dont block the event loop
+          await new Promise(res => {
+            setImmediate(() => {
+              res(null)
+            })
+          })
```

Comment:
you did add util for that:
```suggestion
          await untilNextEventLoopTick()
```
:)

Reply:
Thanks 😅 updated it!